### ISSUE TITLE
App reports "server is in maintenance mode" when trying to download files from external storage with wrong permissions

### ISSUE THREAD

**Issue by wonder75** [nextcloud/android#11504]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

1. Mount external storage in Nextcloud
2. Change permissions on a file so Nextcloud user on server can't read the file (mine was r------ for another user)
3. Try to download the file with the app

### Expected behaviour

Provide a better error message. The problem had nothing to do with the instance being in "maintenance mode".

Maybe something like "Permission problem. Nextcloud can't read this file"

### Actual behaviour

"Download failed. Server is in Maintenance Mode"

### Android version

12

### Device brand and model

Pixel6a

### Stock or custom OS?

Stock

### Nextcloud android app version

3.22.2

### Nextcloud server version

25.0.5

### Using a reverse proxy?

No

**Comment by joshtrichards** [comment]
I reproduced your issue. Does _not_ require external storage. I triggered it easily enough by creating messed up permissions via a shell. 

At the moment the server only returns a general `503 service unavailable` so being more forthcoming about the underlying cause isn't possible within the client. 

The client is displaying its own generic message associated with a 503 http response which assumes maintenance mode (which is one of the main reasons to get a 503). The current message arguably implies a bit too strongly that that is the *only* reason for a 503.

It *might* be possible for the server to include some additional details in the webdav error response. At that point the client could be more informative, but it's debatable what should be disclosed to clients since this issue has nothing to do with them and isn't something they can fix (exception for personal external mounts maybe).

This seems like a bit of an edge case as the permissions issue isn't *in* NC so it can't be fixed by a user but only by an admin generally. The exception *is* getting logged on the server-side for the record so the admin can act if so desired.

I personally like the idea of making the e current message in the client more accurate with regards to circumstances a 503 can be received by it (even when it doesn't have more details to provide the user). Perhaps something like `This request could not be completed at this time. Server may be undergoing maintenance. Retry later or contact your local admin if this persists.`

* https://github.com/nextcloud/server/blob/760952b5083f0a31317721a8c9fe3ed20f89e2e8/apps/dav/lib/Connector/Sabre/File.php#L496
* https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503
